### PR TITLE
Add new certified category views

### PR DESCRIPTION
**Pull request by carkod**
## Done

Added views for each category
Refactored into a single function that dynamically generates the different category search results

## QA

- Check https://ubuntu-com-10114.demos.haus/certified/desktops
- Check https://ubuntu-com-10114.demos.haus/certified/laptops
- Check https://ubuntu-com-10114.demos.haus/certified/socs
- Check https://ubuntu-com-10114.demos.haus/certified/devices
- Check https://ubuntu-com-10114.demos.haus/certified/servers
Or you can also go to the top navigation and try all the links, they should also work.

## Issue / Card

Fixes https://github.com/canonical-web-and-design/web-squad/issues/4286
Fixes https://github.com/canonical-web-and-design/web-squad/issues/4287
Fixes https://github.com/canonical-web-and-design/web-squad/issues/4288
Fixes https://github.com/canonical-web-and-design/web-squad/issues/4289

### CODE REVIEW

**Comment by bethcollins92** [comment]
I'll review this once we've merged mine so we won't need all that javascript on each page 🥳

**Review by bethcollins92 — APPROVED** [review]
LGTM 👍🏼 . Couple of points:

- I'll refactor this with an include for the repeated code as you mentioned, which I can do when I add content to these pages [here](https://github.com/canonical-web-and-design/web-squad/issues/4290).
-  @davegoddard42 would you agree that when we're on the desktop specific page for example, the list of vendors and releases that are displayed in the filter section should only include ones which come under the desktop category? I feel we shouldn't have them in the list if there aren't any desktops by that vendor/release. (Out of scope for this PR but can make another issue)

**Comment by davegoddard42** [comment]
@bethcollins92 yes, that would make sense.

**Comment by carkod** [comment]
> @bethcollins92 yes, that would make sense.

Created issue https://github.com/canonical-web-and-design/web-squad/issues/4317

File: templates/certified/desktops.html
Comment (anthonydillon) [inline templates/certified/desktops.html:3]: Could this be "Certified desktops"

File: templates/certified/desktops.html
Comment (anthonydillon) [inline templates/certified/desktops.html:5]: Are there copy docs for these pages? I feel a better description would be something like: "Ubuntu certifies thousands of desktops to guarantee they work well with Ubuntu." Gives more of a purpose of the page and content.

File: templates/certified/desktops.html
Comment (anthonydillon) [inline templates/certified/desktops.html:170]: Maybe "Sorry we found on results matching your criteria. Please try and widen your search."

File: templates/certified/desktops.html
Comment (anthonydillon) [inline templates/certified/desktops.html:177]: `p` tags used within the `lil` will give you the spacing you are looking for without `u-sv1`

File: templates/certified/devices.html
Comment (anthonydillon) [inline templates/certified/devices.html:3]: Same comments as the above document.

File: templates/certified/devices.html
Comment (anthonydillon) [inline templates/certified/devices.html:5]: Try and find some wording that defines the purpose of the content.

File: templates/certified/laptops.html
Comment (anthonydillon) [inline templates/certified/laptops.html:5]: Same here

File: templates/certified/servers.html
Comment (anthonydillon) [inline templates/certified/servers.html:5]: Same here

File: templates/certified/socs.html
Comment (anthonydillon) [inline templates/certified/socs.html:5]: Same here

File: templates/certified/desktops.html
Comment (bethcollins92) [reply templates/certified/desktops.html:5]: FYI Dave is still chasing the PM for copy for these pages. There are rough drafts [here](https://drive.google.com/drive/folders/1a_Iz5iw7nt4VeInmHoACfCihW7ZU83bI) but they need looking at again and also need copy for the vendor specific pages

Comment (davegoddard42) [reply templates/certified/desktops.html:5]: I am actively chasing - spoke to the PM today. I've created a template for the vendor page too

File: templates/certified/devices.html
Comment (carkod) [reply templates/certified/devices.html:5]: I would rather leave this to the copy docs / PM, whatever I come up with will most likely be replaced soon.

**Pull request merged**
